src/Google/GlobalSiteTag.php

```diff
@@ -602,4 +606,118 @@ private function register_js_for_fast_refresh_dev() {
 			false
 		);
 	}
+
+	/**
+	 * Set user data config when Enhanced Conversions is enabled.
+	 *
+	 * @return string|null
+	 */
+	public function get_enhanced_conversion_tag() {
+		// TODO: Check enhanced conversion option status when implemented.
+		// $enhanced_conversions =$this->options->get( OptionsInterface::ADS_ENHANCED_CONVERSION_STATUS );
+		$enhanced_conversions = true;
+
+		if ( ! $enhanced_conversions ) {
+			return;
+		}
+
+		// Retrieve user data from the current session.
+		$customer = WC()->session->get( 'customer' );
+
+		$ec_data = [];
+
+		// Add email address to enhanced conversion data.
+		if ( ! empty( $customer['email'] ) ) {
+			$ec_data['sha256_email_address'] = esc_js( $this->normalize_and_hash( $customer['email'] ) );
```

Comment:
I don't think we need `esc_js` here or throughout, since all of this data will be passed to `wp_json_encode`, unless there is something specifically you are using this for.

FWIW, I generally consider `esc_js` to be a soft-deprecated function in WP Core at this point. Note [this comment from Weston](https://developer.wordpress.org/reference/functions/esc_js/#comment-1747) as to why it's no longer considered best practice.